### ISSUE TITLE
Terraform apply with a prior plan seems to ignore backend configuration

### ISSUE THREAD

**Issue by makered** [hashicorp/terraform#12821]
### Terraform Version
Terraform v0.9.0

### Affected Resource(s)
core

### Terraform Configuration Files
```hcl
terraform {
  required_version = ">= 0.9.0"
  backend "s3" {
    bucket = "mybucket"
    key    = "mypath/terraform.tfstate"
    region = "us-east-1"
  }
}
```

### Expected Behavior
Running `terraform plan -out=plan.out` followed by `terraform apply plan.out`
should follow the backend configuration and update the state in S3

### Actual Behavior
Running `terraform plan -out=plan.out` followed by `terraform apply plan.out`
creates a local `terraform.tfstate` and does NOT update state in S3

### Steps to Reproduce
1. Configure S3 as the backend
2. `terraform plan -out=plan.out`
3. `terraform apply plan.out`

**Comment by mitchellh** [comment]
Yes it does we just fixed this and it will be in 0.9.1 :D Sorry!

**Comment by makered** [comment]
Thanks for super fast response and fix!
